### PR TITLE
Add guide on how to find project number and column id for projects classic and V2

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**

### CODE REVIEW

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:27]: For (classic), just copy column link. And use the end of URL

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:65]: If I would be hunting classic project column ID, the guide tells me to start with creating token that is not needed.
The titles should follow the GitHub structure. Something like
```
..intro..
# Projects (classic)

## Find column id

# Projects (V2)

## Find project number

## Find column ID

### ...token
### ...query


```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:35]:
```suggestion
### Find the column ID
```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:7]: This should not be needed. The token will not push YML files in `.github/workflow`
```suggestion
```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:8]: It doesn't query user, so it should not be needed
```suggestion
```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:6]: This should not be needed, it doesn't query anything from repo
```suggestion
```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:25]: Can you make it a single copy/paste `curl` command with headers and post content? That should be available on both, windows and UNIX the same way. It would also merge this and the query at the bottom into a single section that will be easy to use and follow

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:31]:
```suggestion
The project number is available in the URL of your project as: https://github.com/orgs/SonarSource/projects/<NUMBER>
```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:33]: I'm not sure what this is for. The URL above is pretty clear
```suggestion
```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:43]: This would be confusing. Nobody will need this in YML
```suggestion
```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:40]: The parametrization makes it very complicated. I'd hardcode it instead. And the org doesn't can be hardcoded to SonarSource

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
Round 1 :) I'll need to do a full second review after the 1st batch is done

File: docs/github.md
Comment (ilia-kebets-sonarsource) [reply docs/github.md:60]: ACK

File: docs/github.md
Comment (ilia-kebets-sonarsource) [reply docs/github.md:3]: will do!

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:30]: This is already written above
```suggestion
```

File: docs/github.md
Comment (ilia-kebets-sonarsource) [reply docs/github.md]: I prefer to keep it as it is the kind of link that I sometimes took some time to find. I changed the format.

File: docs/github.md
Comment (ilia-kebets-sonarsource) [reply docs/github.md]: I've fixed the first header with double quotes `"`, I tested it on a colleague's Windows machine. Not sure you have the same setup.

Comment (ilia-kebets-sonarsource) [reply docs/github.md]: what is your cURL version?

Comment (pavel-mikula-sonarsource) [reply docs/github.md]: curl 8.0.1 (Windows) libcurl/8.0.1 Schannel WinIDN
Release-Date: 2023-03-20
Protocols: dict file ftp ftps http https imap imaps pop3 pop3s smtp smtps telnet tftp
Features: AsynchDNS HSTS HTTPS-proxy IDN IPv6 Kerberos Largefile NTLM SPNEGO SSL SSPI threadsafe Unicode UnixSockets

Comment (pavel-mikula-sonarsource) [reply docs/github.md]: The current version still doesn't work. Based on this https://mkyong.com/web/curl-post-json-data-on-windows/ you'll need to use double quotes about the POST json and escape the inner quotes.

Comment (ilia-kebets-sonarsource) [reply docs/github.md]: I'm a 400 `Problems parsing JSON` error when using double quotes on `-d` like this:

```curl
curl -H "Content-Type: application/json" -H "Authorization: bearer <YOUR_GITHUB_PERSONAL_ACCESS_TOKEN>" -X POST -d "{ \"query\": \"query {  organization(login: \\"SonarSource\\") { projectV2(number: <NUMBER>) { field(name: \\"Status\\") { ... on ProjectV2SingleSelectField { columns: options { id name }}}}}}\" }" https://api.github.com/graphql
```

I'm not sure how to escape the inner quotes for `SonarSource` or `Status`.

However, I've found the usage of `--data-raw` in [MS docs](https://learn.microsoft.com/en-us/azure/developer/javascript/how-to/with-web-app/graphql/static-web-app-graphql/graphql-basics#http-curl-request). Would you try it?

```curl
curl -H "Content-Type: application/json" -H "Authorization: bearer <YOUR_GITHUB_PERSONAL_ACCESS_TOKEN>" -X POST --data-raw '{ "query": "query {  organization(login: \"SonarSource\") { projectV2(number: <NUMBER>) { field(name: \"Status\") { ... on ProjectV2SingleSelectField { columns: options { id name }}}}}}" }' https://api.github.com/graphql
```

Comment (pavel-mikula-sonarsource) [reply docs/github.md]: --data-raw is not going to help, because it's the console that fails to parse it.

Let's just give up and split it into 2 commands. Making one for both platforms seems too difficult.

Comment (ilia-kebets-sonarsource) [reply docs/github.md]: Do you mean splitting into different commands for Windows and MacOS?

File: docs/github.md
Comment (ilia-kebets-sonarsource) [reply docs/github.md:30]: it works on MacOS and Windows/Cygwin

Comment (pavel-mikula-sonarsource) [reply docs/github.md:30]: Dumping an idea:
For the future, we should probably leave the `column-id` idea and require `project-number` and `column-name` everywhere, to make it easier to work with. 
I'm not sure how we would distinguish between V1 and V2 yet.

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:26]: EOL and Q
```suggestion

#### Query
```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:28]: To be extra clear
```suggestion
To get the column ids, use the following command, replacing `<YOUR_GITHUB_PERSONAL_ACCESS_TOKEN>` and `<NUMBER>`:
```

File: docs/github.md
Comment (pavel-mikula-sonarsource) [inline docs/github.md:23]: Let's swap these, as that's the order in GitHub UI
```suggestion
- `public_repo`
- `read:project`
```

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM, last 3 polishing comments before merging

**Pull request merged**
